x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/action_log_expanded_tray.tsx

```diff
@@ -0,0 +1,221 @@
+  paddingSize: 'none',
+})`
+  code {
+    color: ${(props) => props.theme.eui.euiColorDarkShade} !important;
+  }
+`;
+
+export const ActionsLogExpandedTray = memo<{
+  action: MaybeImmutable<ActionDetails>;
+}>(({ action }) => {
+  const getTestId = useTestIdGenerator('response-actions-list');
+  const { canWriteFileOperations, canWriteExecuteOperations } =
+    useUserPrivileges().endpointPrivileges;
+
+  const {
+    startedAt,
+    completedAt,
+    isCompleted,
+    wasSuccessful,
+    isExpired,
+    command: _command,
+    comment,
+    parameters,
+  } = action;
+
+  const parametersList = parameters
+    ? Object.entries(parameters).map(([key, value]) => {
+        return `${key}:${value}`;
+      })
+    : undefined;
+
+  const command = getUiCommand(_command);
+  const isGetFileCommand = command === 'get-file';
+  const isExecuteCommand = command === 'execute';
+  const dataList = [
+    {
+      title: OUTPUT_MESSAGES.expandSection.placedAt,
+      description: `${startedAt}`,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.startedAt,
+      description: `${startedAt}`,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.completedAt,
+      description: `${completedAt ?? emptyValue}`,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.input,
+      description: `${command}`,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.parameters,
+      description: parametersList ? parametersList : emptyValue,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.comment,
+      description: comment ? comment : emptyValue,
+    },
+  ].map(({ title, description }) => {
+    return {
+      title: <StyledEuiCodeBlock>{title}</StyledEuiCodeBlock>,
+      description: <StyledEuiCodeBlock>{description}</StyledEuiCodeBlock>,
+    };
+  });
+
+  const getOutputContent = () => {
+    if (isExpired) {
+      return OUTPUT_MESSAGES.hasExpired(command);
+    }
+
+    if (!isCompleted) {
+      return OUTPUT_MESSAGES.isPending(command);
+    }
+
+    if (!wasSuccessful) {
+      return OUTPUT_MESSAGES.hasFailed(command);
+    }
+
+    if (isGetFileCommand) {
+      return (
+        <>
+          {OUTPUT_MESSAGES.wasSuccessful(command)}
+          <ResponseActionFileDownloadLink
+            action={action}
+            canAccessFileDownloadLink={canWriteFileOperations}
+            textSize="xs"
+            data-test-subj={getTestId('getFileDownloadLink')}
+          />
+        </>
+      );
+    }
+
+    if (isExecuteCommand) {
+      return (
+        <EuiFlexGroup direction="column" data-test-subj={getTestId('executeDetails')}>
+          {action.agents.map((agentId) => (
+            <div key={agentId}>
+              {OUTPUT_MESSAGES.wasSuccessful(command)}
+              <EuiFlexItem>
+                <ResponseActionFileDownloadLink
+                  action={action}
+                  buttonTitle={EXECUTE_FILE_LINK_TITLE}
+                  canAccessFileDownloadLink={canWriteExecuteOperations}
```

Comment:
can we check on this with Caitlyn + Kevin. For `get-file` we explicitly said that they should only gain access to the file content (download) from the History Log if they had permissions to it (we did not want to expose data to those that don't have access to it). Lets confirm that should also be the case for `execute` (IMO, it probably should not, but lets pose the question)

Reply:
Yeah, I went with this cause it makes sense to keep RBAC privileges separate. @kevinlog @caitlinbetz thoughts? Should we restrict file download link from the actions log if they don't have explicit permission to it (no file operations or no execute operations)

Reply:
No, I have not yet.